src/reducers/shelf/filter.ts

```diff
@@ -145,3 +186,54 @@ export function getFilter(fieldDef: ShelfFieldDef, domain: any[]): RangeFilter |
       throw new Error('Unsupported type ' + fieldDef.type);
   }
 }
+
+export function getAllTimeUnits() {
+  return [
+    TimeUnit.YEARMONTHDATE,
+    TimeUnit.YEAR,
+    TimeUnit.MONTH,
+    TimeUnit.QUARTER,
+    TimeUnit.DATE,
+    TimeUnit.DAY,
+    TimeUnit.HOURS,
+    TimeUnit.MINUTES,
+    TimeUnit.SECONDS,
+    TimeUnit.MILLISECONDS
+  ];
+}
+
+export function getDefaultRange(domain: number[], timeUnit: TimeUnit): number[] {
+  switch (timeUnit) {
+    case TimeUnit.YEARMONTHDATE:
+      return [Number(convert(timeUnit, new Date(domain[0]))),
+        Number(convert(timeUnit, new Date(domain[1])))];
+    case TimeUnit.YEAR:
+      return [convert(timeUnit, new Date(domain[0])).getFullYear(),
+        convert(timeUnit, new Date(domain[1])).getFullYear()];
+    case TimeUnit.QUARTER:
+      return [1, 4];
+    case TimeUnit.DATE:
+      return [1, 31];
+    case TimeUnit.HOURS:
+      return [0, 23];
+    case TimeUnit.MINUTES:
+      return [0, 59];
+    case TimeUnit.SECONDS:
+      return [0, 59];
+    case TimeUnit.MILLISECONDS:
+      return [0, 999];
+    default:
+      throw new Error ('Invalid range time unit ' + timeUnit);
+  }
+}
+
+export function getDefaultList(timeUnit: TimeUnit): number[] {
+  switch (timeUnit) {
+    case TimeUnit.MONTH:
+      return [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    case TimeUnit.DAY:
+      return [1, 2, 3, 4, 5, 6, 7];
```

Comment:
Do we have an issue filed for changing label for MONTH and DAY? 

(Month is not that confusing, but I have no idea what 1 means for DAY (Sunday / Monday?)